pkg/configmap/configmap_watcher.go

```diff
@@ -0,0 +1,177 @@
+	cm, err := cmw.clientset.CoreV1().
+		ConfigMaps(cmw.namespace).Get(ctx, configMapName, v1.GetOptions{})
+	if err != nil {
+		return config.Config{}, fmt.Errorf("getting initial configuration: %w", err)
+	}
+
+	return cmw.getKubeLinterConfig(cm.Data[configMapDataAccess])
+}
+
+// Start will update the channel structure with new configuration data from ConfigMap update event
+func (cmw Watcher) Start(ctx context.Context) error {
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		cmw.clientset, time.Second*30, informers.WithNamespace(cmw.namespace),
+	)
+	informer := factory.Core().V1().ConfigMaps().Informer()
+
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{ // nolint:errcheck
+		AddFunc: func(obj interface{}) {
+			newCm := obj.(*apicorev1.ConfigMap)
+
+			if configMapName != newCm.GetName() {
+				return
+			}
+
+			cmw.logger.Info("ConfigMap has been created")
+
+			cfg, err := cmw.getKubeLinterConfig(newCm.Data[configMapDataAccess])
+			if err != nil {
+				cmw.logger.Error(err, "ConfigMap data format")
+				return
+			}
+
+			cmw.ch <- cfg
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			newCm := newObj.(*apicorev1.ConfigMap)
+
+			// This is sometimes triggered even if no change was due to the ConfigMap
+			if configMapName != newCm.GetName() || reflect.DeepEqual(oldObj, newObj) {
+				return
+			}
+
+			cmw.logger.Info("ConfigMap has been updated")
+
+			cfg, err := cmw.getKubeLinterConfig(newCm.Data[configMapDataAccess])
+			if err != nil {
+				cmw.logger.Error(err, "ConfigMap data format")
+				return
+			}
+
+			cmw.ch <- cfg
+		},
+	})
+
+	factory.Start(ctx.Done())
+
+	return nil
+}
+
+// ConfigChanged receives push notifications when the configuration is updated
+func (cmw *Watcher) ConfigChanged() <-chan config.Config {
+	return cmw.ch
+}
+
+// getKubeLinterConfig returns a valid Kube-linter Config structure
+// based on the checks received by the string
+func (cmw *Watcher) getKubeLinterConfig(data string) (config.Config, error) {
+	var cfg config.Config
+
+	err := yaml.Unmarshal([]byte(data), &cmw.checks)
+	if err != nil {
+		return cfg, fmt.Errorf("unmarshalling configmap data: %w", err)
+	}
+
+	cfg.Checks = config.ChecksConfig(cmw.checks.Checks)
+
+	return cfg, nil
+}
+
+// getDeploymentNamespace retrieves the namespace of the Deployment associated with DVO.
+// If found, it returns the namespace of the Deployment.
+// If not found, it returns the default namespace value from `dvoConfig.OperatorNamespace`.
+func getDeploymentNamespace(clientset *kubernetes.Clientset) (string, error) {
+	deployments, err := clientset.AppsV1().Deployments(v1.NamespaceAll).
+		List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("getting deployments from clientset: %w", err)
+	}
+
+	for _, deployment := range deployments.Items {
+		if deployment.GetName() == dvoConfig.OperatorName {
+			return deployment.GetNamespace(), nil
+		}
+	}
```

Comment:
I would have other suggestion (sorry maybe I am too picky). There's `POD_NAME` env variable available for the DVO container. We can do `os.Getenv("POD_NAME")` and then get the Pod and check its namespace. I know, I know...I was talking about deployment, but this is probably better and safer way to get it.